pkg/controller/eventlog/eventlog.go

```diff
@@ -0,0 +1,52 @@
+package eventlog
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ControllerWithEvents interface {
+	EventRecorder() *record.EventRecorder
+	Logger() *logr.Logger
+	Client() *client.Client
+}
+
+type ControllerWithLogger interface {
```

Comment:
Is ther any advantage in having `ControllerWithLogger` and `ControllerWithEvents` as separate interfaces? It seems like all controller satisfy both interfaces. How about having a single interface `EventLogger`:

```go
type EventLogger interface {
    EventRecorder() *record.EventRecorder
    Logger() *logr.Logger
}
```

Reply:
seems that `batchController` doesn't implement the event part

Reply:
IMO it's better to add the EventRecorder to the batch_controller now, as I'm sure that utilising events for batch could be useful, instead of splitting into two interfaces for this one case.

Reply:
fixed